langchain/src/chat_models/base.ts

```diff
@@ -0,0 +1,864 @@
+import {
+  BaseLanguageModelInput,
+  ToolDefinition,
+} from "@langchain/core/language_models/base";
+import {
+  BaseChatModel,
+  type BaseChatModelCallOptions,
+} from "@langchain/core/language_models/chat_models";
+import { BaseMessage, type AIMessageChunk } from "@langchain/core/messages";
+import {
+  Runnable,
+  type RunnableBatchOptions,
+  RunnableBinding,
+  type RunnableConfig,
+  type RunnableToolLike,
+  ensureConfig,
+} from "@langchain/core/runnables";
+import {
+  AsyncGeneratorWithSetup,
+  IterableReadableStream,
+} from "@langchain/core/utils/stream";
+import {
+  type LogStreamCallbackHandlerInput,
+  type RunLogPatch,
+  type StreamEvent,
+} from "@langchain/core/tracers/log_stream";
+import { type StructuredToolInterface } from "@langchain/core/tools";
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import { ChatResult } from "@langchain/core/outputs";
+
+// TODO: remove once `EventStreamCallbackHandlerInput` is exposed in core.
+interface EventStreamCallbackHandlerInput
+  extends Omit<LogStreamCallbackHandlerInput, "_schemaFormat"> {}
+
+export type ChatModelProvider =
+  | "openai"
+  | "anthropic"
+  | "azure_openai"
+  | "cohere"
+  | "google_vertexai"
+  | "google_genai"
+  | "ollama"
+  | "together"
+  | "fireworks"
+  | "mistralai"
+  | "groq"
+  | "bedrock";
```

Comment:
You don't need to list these twice - something like `export type ChatModelProvider = typeof keyof _SUPPORTED_PROVIDERS` should work